src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -163,7 +249,18 @@ struct ExeSpaceUtils<Serialize,Kokkos::Cuda> {
                         const Lambda& lambda,
                         ValueType& result)
   {
-    impl::parallel_reduce<TeamMember, Lambda, ValueType, Serialize>(team, begin, end, lambda, result);
+    impl::parallel_reduce<Serialize, TeamMember, Lambda, ValueType>(team, begin, end, lambda, result);
+  }
+
+  template <typename TeamMember, typename InputProvider, typename ValueType>
+  static KOKKOS_INLINE_FUNCTION
+  void view_reduction (const TeamMember& team,
+                       const int& begin,
```

Comment:
Since `begin` and `end` refer variously to pack and scalar, I recommend an inline comment in each place like this: `const int& begin, // scalar index` or `const int& begin, // pack index`.

Reply:
In the `impl` version, too, please, where in particular it's `pack index` rather than `scalar index`.